lightning/src/ln/channelmanager.rs

```diff
@@ -6122,14 +6122,50 @@ impl<Signer: Sign, M: Deref , T: Deref , K: Deref , F: Deref , L: Deref >
 	}
 
 	fn provided_node_features(&self) -> NodeFeatures {
-		NodeFeatures::known()
+		provided_node_features()
 	}
 
 	fn provided_init_features(&self, _their_init_features: &PublicKey) -> InitFeatures {
-		InitFeatures::known_channel_features()
+		provided_init_features()
 	}
 }
 
+/// Fetches the set of [`NodeFeature`] flags which are provided by or required by
+/// [`ChannelManager`].
+pub fn provided_node_features() -> NodeFeatures {
+	provided_init_features().to_context()
+}
+
+/// Fetches the set of [`InvoiceFeature`] flags which are provided by or required by
+/// [`ChannelManager`].
+pub fn provided_invoice_features() -> InvoiceFeatures {
+	provided_init_features().to_context()
```

Comment:
Do we care that invoice-only features like `option_payment_metadata` won't be returned?